src/promptflow-tools/tests/test_common.py

```diff
@@ -119,37 +258,250 @@ def test_success_parse_role_prompt(self, chat_str, images, expected_result):
     @pytest.mark.parametrize(
         "chat_str, expected_result",
         [
-            ("\n#system:\n##name:\nAI \n content:\nfirst\n\n#user:\nsecond", [
-                {'role': 'system', 'name': 'AI', 'content': 'first'}, {'role': 'user', 'content': 'second'}]),
-            ("\nuser:\nname:\n\nperson\n content:\n", [
-                {'role': 'user', 'name': 'person', 'content': ''}]),
-            ("\nsystem:\nname:\n\n content:\nfirst", [
-                {'role': 'system', 'content': 'name:\n\n content:\nfirst'}]),
-            ("\nsystem:\nname:\n\n", [
-                {'role': 'system', 'content': 'name:'}])
-        ]
+            (
+                "\n#system:\n##name:\nAI \n content:\nfirst\n\n#user:\nsecond",
+                [{"role": "system", "name": "AI", "content": "first"}, {"role": "user", "content": "second"}],
+            ),
+            ("\nuser:\nname:\n\nperson\n content:\n", [{"role": "user", "name": "person", "content": ""}]),
+            ("\nsystem:\nname:\n\n content:\nfirst", [{"role": "system", "content": "name:\n\n content:\nfirst"}]),
+            ("\nsystem:\nname:\n\n", [{"role": "system", "content": "name:"}]),
+        ],
     )
     def test_parse_chat_with_name_in_role_prompt(self, chat_str, expected_result):
         actual_result = parse_chat(chat_str)
         assert actual_result == expected_result
 
+    @pytest.mark.parametrize(
+        "chat_str, expected_result, success, error_message",
+        [
+            (
+                '# tool:\n## tool_call_id:\ncall_001\n## content:\n{"location": "San Francisco, CA", "temperature": "72", "unit": null}\n',
+                [
+                    {
+                        "role": "tool",
+                        "tool_call_id": "call_001",
+                        "content": '{"location": "San Francisco, CA", "temperature": "72", "unit": null}',
+                    }
+                ],
+                True,
+                "",
+            ),
+            (
+                """
+# tool:
```

Comment:
save in a separate file for this complicate test case.

Reply:
updated